sdk/servicebus/service-bus/src/serviceBusMessageBatch.ts

```diff
@@ -0,0 +1,190 @@
+   */
+  private _maxSizeInBytes: number;
+  /**
+   * @property Current size of the batch in bytes.
+   */
+  private _sizeInBytes: number;
+  /**
+   * @property Encoded amqp messages.
+   */
+  private _encodedMessages: Buffer[] = [];
+  /**
+   * @property Number of messages in the batch.
+   */
+  private _count: number;
+  /**
+   * @property Encoded batch message.
+   */
+  private _batchMessage: Buffer | undefined;
+
+  /**
+   * ServiceBusMessageBatch should not be constructed using `new ServiceBusMessageBatch()`
+   * Use the `createBatch()` method on your `Sender` instead.
+   * @constructor
+   * @internal
+   * @ignore
+   */
+  constructor(context: ClientEntityContext, maxSizeInBytes: number) {
+    this._context = context;
+    this._maxSizeInBytes = maxSizeInBytes;
+    this._sizeInBytes = 0;
+    this._count = 0;
+  }
+
+  /**
+   * @property The maximum size of the batch, in bytes.
+   * @readonly
+   */
+  get maxSizeInBytes(): number {
+    return this._maxSizeInBytes;
+  }
+
+  /**
+   * @property Size of the `ServiceBusMessageBatch` instance after the messages added to it have been
+   * encoded into a single AMQP message.
+   * @readonly
+   */
+  get sizeInBytes(): number {
+    return this._sizeInBytes;
+  }
+
+  /**
+   * @property Number of messages in the `ServiceBusMessageBatch` instance.
+   * @readonly
+   */
+  get count(): number {
+    return this._count;
+  }
+
+  /**
+   * @property Represents the single AMQP message which is the result of encoding all the events
+   * added into the `ServiceBusMessageBatch` instance.
+   *
+   * This is not meant for the user to use directly.
+   *
+   * When the `ServiceBusMessageBatch` instance is passed to the `sendBatch()` method on the `Sender`,
+   * this single batched AMQP message is what gets sent over the wire to the service.
+   * @readonly
+   */
+  get _message(): Buffer | undefined {
+    return this._batchMessage;
+  }
+
+  /**
+   * Tries to add a message to the batch if permitted by the batch's size limit.
+   * **NOTE**: Always remember to check the return value of this method, before calling it again
+   * for the next message.
+   *
+   * @param message  An individual service bus message.
+   * @returns A boolean value indicating if the message has been added to the batch or not.
+   */
+  public tryAdd(message: ServiceBusMessage): boolean {
+    throwTypeErrorIfParameterMissing(this._context.namespace.connectionId, "tryAdd", "message");
+
+    // Convert ServiceBusMessage to AmqpMessage.
+    const amqpMessage = toAmqpMessage(message);
+    amqpMessage.body = this._context.namespace.dataTransformer.encode(message.body);
+
+    // Encode every amqp message and then convert every encoded message to amqp data section
+    this._encodedMessages.push(RheaMessageUtil.encode(amqpMessage));
+
+    const batchMessage: AmqpMessage = {
+      body: RheaMessageUtil.data_sections(this._encodedMessages)
+    };
+
+    if (amqpMessage.message_annotations) {
```

Comment:
I suppose we can just assign this directly? In the worst case we assign an undefined value, which is what it would have been anyways?

Reply:
Sounds good. I was just following the older `sendArrayOfMessages` and doing the same here.
Removed the unnecessary if checks. :)